### ISSUE TITLE
class generated between compiles is different

### ISSUE THREAD

**Issue by pouloghost** [vimeo/stag-java#181]
#### Issue Summary
generated classes depend on         
`Set<? extends Element> useStagElements = roundEnv.getElementsAnnotatedWith(UseStag.class);`
can vary between compiles, making hotfix very hard.

#### Reproduction Steps
just compile several times

#### Expected Behavior
generated classes stay the same unless code changes.

#### Fix step
`
       Set<? extends Element> useStagElements = roundEnv.getElementsAnnotatedWith(UseStag.class);
        List<? extends Element> sorted = new ArrayList<>(useStagElements);
        sorted.sort(new Comparator<Element>() {
            @Override
            public int compare(Element element, Element t1) {
                return element.toString().compareTo(t1.toString());
            }
        });
`

**Comment by anthonycr** [comment]
Great find, I dug into the code and discovered it is due to the usage of `HashSet` and `HashMap` in the code. Neither of these collections guarantee iteration order, and therefore cause builds to be non-deterministic. The fix was to switch to `LinkedHashSet` and `LinkedHashMap` for all usages of `Set` and `Map` that were iterated.

I did not perform sorting of the elements returned by `getElementsAnnotatedWith` as it is deterministic and returns the elements in a fixed order for the same input. I verified this additionally by adding print statements of `useStagElements.toString().hashCode()`, and observed that it printed the same value for every invocation of `./gradlew clean :sample:assembleDebug`.

Bug fixed here: https://github.com/vimeo/stag-java/pull/184
